Tree/Processing/PostProcessing/SecurityProcessor.php

```diff
@@ -0,0 +1,160 @@
+{
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+
+    /**
+     * @var ExpressionLanguage
+     */
+    private $language;
+
+
+    /**
+     * @var AuthenticationTrustResolverInterface
+     */
+    private $trustResolver;
+
+
+    /**
+     * @var RoleHierarchyInterface
+     */
+    private $roleHierarchy;
+
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+
+
+    /**
+     * @param TokenStorageInterface                $tokenStorage
+     * @param AuthorizationCheckerInterface        $authorizationChecker
+     * @param ExpressionLanguage                   $language
+     * @param AuthenticationTrustResolverInterface $trustResolver
+     * @param RoleHierarchyInterface               $roleHierarchy
+     * @param RequestStack                         $requestStack
+     */
+    public function __construct (TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authorizationChecker, ExpressionLanguage $language, AuthenticationTrustResolverInterface $trustResolver, RoleHierarchyInterface $roleHierarchy, RequestStack $requestStack)
+    {
+        $this->tokenStorage = $tokenStorage;
+        $this->authorizationChecker = $authorizationChecker;
+        $this->language = $language;
+        $this->trustResolver = $trustResolver;
+        $this->roleHierarchy = $roleHierarchy;
+        $this->requestStack = $requestStack;
+    }
+
+
+
+    /**
+     * @param Node $node
+     *
+     * @return bool
+     */
+    private function isAllowedToAccessNode (Node $node)
+    {
+        if (empty($node->getSecurity()))
+        {
+            // always allowed, if no security is set
+            return true;
+        }
+
+        // if the user is not behind a firewall but a security string is set, prevent access
+        if (null === $this->tokenStorage->getToken())
+        {
+            return false;
+        }
+
+        return $this->language->evaluate($node->getSecurity(), $this->getVariables());
+    }
+
+
+
+    /**
+     * Returns the variables for the expression evaluation
+     *
+     * code should be sync with Symfony\Component\Security\Core\Authorization\Voter\ExpressionVoter
+     *
+     * @return array
+     */
+    private function getVariables ()
+    {
+        $request = $this->requestStack->getMasterRequest();
+        $requestVariables = null !== $request
+            ? $request->attributes->all()
+            : [];
+        $token = $this->tokenStorage->getToken();
+
+        /** @var Role[] $roles */
+        $roles = null !== $this->roleHierarchy
+            ? $this->roleHierarchy->getReachableRoles($token->getRoles())
+            : $token->getRoles();
+
+        $variables = [
+            'token' => $token,
+            'user' => $token->getUser(),
+            'request' => $request,
+            'roles' => array_map(function ($role)
```

Comment:
I thought that was the original layout, but it was automatically layouted by my Phpstorm. I think it would be wise to keep the original layout here (from `Sensio\Bundle\FrameworkExtraBundle\EventListener\SecurityListener`), so it is one line.